include/aspect/material_model/interface.h

```diff
@@ -359,9 +359,10 @@ namespace aspect
       template <int dim>
       void average (const AveragingOperation operation,
                     const typename DoFHandler<dim>::active_cell_iterator &cell,
-                    const Quadrature<dim>   &quadrature_formula,
-                    const Mapping<dim>      &mapping,
-                    MaterialModelOutputs    &values);
+                    const Quadrature<dim>     &quadrature_formula,
+                    const Mapping<dim>        &mapping,
+                    MaterialModelOutputs      &values_out,
+                    const MaterialModelInputs<dim>  &values_in);
     }
```

Comment:
Having the output argument in the middle is somewhat awkward. I would move ```values_out``` to the beginning. Or do you want to keep it at the end, @bangerth ?